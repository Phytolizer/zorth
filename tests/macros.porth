include "std.porth"

macro check_less
  < if
    "YES\n" stdout write
  else
    "NO\n" stdout write
  end
end

1 2 check_less
2 1 check_less

macro even_fibs
  0 1 while over 1000000 < do
    over 2 mod 0 = if
      over print
    end
    swap over +
  end
end

"------------------------------\n" stdout write

even_fibs
