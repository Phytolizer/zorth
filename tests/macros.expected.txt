: returncode 0
: stdout 86
YES
NO
------------------------------
0
2
8
34
144
610
2584
10946
46368
196418
832040

: stderr 0

